plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -102,8 +103,9 @@ public boolean originatedFromDockerExplorer() {
 
 	@Override
 	public IDockerConnection getDockerConnection() {
-		if(dockerConnection == null && getDockerConnections().size() == 1) {
-			setDockerConnection(getDockerConnections().get(0));
+		List<IDockerConnection> all = getDockerConnections();
+		if(dockerConnection == null && all.size() == 1) {
+			setDockerConnection(all.get(0));
 		}
```

Comment:
Previous way was better in not calling getDockerConnections() when dockerConnection == null.
Taking the best from both,

if(dockerConnection == null) {
    List<IDockerConnection> all = getDockerConnections();
    if(all.size() == 1) {
        setDockerConnection(all.get(0));
    }
}